R/onlineCPD.R

```diff
@@ -113,18 +113,32 @@ function(datapts, oCPD=NULL,missPts = "none",
   if(is.null(ocpd_settings$cpthreshold)) ocpd_settings$cpthreshold<- cpthreshold
   if(is.null(ocpd_settings$truncRlim)) ocpd_settings$truncRlim<- truncRlim
 
-  # if loading an existing ocpd object, check for compatible settings
-  if(!is.null(oCPD)){
-    if(!(class(oCPD)=="ocp")) stop("Argument oCPD must be of type \"ocp\"")
-    if(!identical(as.character(oCPD$ocpd_settings[c(-1, -2)]), as.character(ocpd_settings[c(-1, -2)])))
-      stop(print("Incompatible ocpd settings with input ocpd object."),
-           for(set_id in 1:length(ocpd_settings)){ # starting from "misspts"
-             if(as.character(ocpd_settings[[set_id]])!= as.character(oCPD$ocpd_settings[[set_id]])){
-               print(ocpd_settings[[set_id]])
-               print(oCPD$ocpd_settings[[set_id]])
-             }
-           })
-  }
+   # if loading an existing ocpd object, check for compatible settings
+    if(!is.null(oCPD)){
+      if(!(class(oCPD)=="ocp")) stop("Argument oCPD must be of type \"ocp\"")
+      param_to_check <- c("probModel","init_params","hazard_func","cpthreshold","truncRlim")
```

Comment:
thanks for the contributions
for this list, it would be better to include all parameters that could affect the calculations, add: missPts, cpthreshold, truncRlim, minRlength, maxRlength, minsep, maxsep, getR